File: src/main/java/problems/Uri1120_RevisaoContrato.java
package problems;

import java.io.IOException;
import java.util.Scanner;

public class Uri1120_RevisaoContrato {
    public static void main(String[] args) throws IOException {
        Scanner ler = new Scanner(System.in);
        long j;
        String numero;

        while (true) {
            j = ler.nextLong();
            numero = ler.next();

            if (j == 0 || numero.equals("0")) {
                break;
            }
            numero = numero.replace(j + "", "");

            for (int i = 0; i < numero.length(); i++) {
                if (numero.charAt(i) == '0') {
                    numero = numero.substring(i, numero.length() - 1);
                }
            }

            if (numero.trim().equals("")) {
                numero = "0";
            }

            System.out.println((numero));
        }
    }
}

File: src/main/kotlin/problems/Uri1120_RevisaoContrato.kt
package problems

import java.util.Scanner

fun main(args: Array<String>) {
    val ler = Scanner(System.`in`)
    while (true) {
        val digitoErrado = ler.next()
        var numero = ler.next()
        if (digitoErrado == "0" || numero == "0")
            break

        numero = numero.replace(digitoErrado, "")

        if (numero.isEmpty())
            println(0)
        else println(numero.toBigInteger())
    }
}